src/mango_httpd.erl

```diff
@@ -50,9 +50,27 @@ handle_req_int(_, _) ->
 
 
 handle_index_req(#httpd{method='GET', path_parts=[_, _]}=Req, Db) ->
+    Params = lists:flatmap(fun({K, V}) -> parse_index_param(K, V) end,
+        chttpd:qs(Req)),
     Idxs = lists:sort(mango_idx:list(Db)),
-    JsonIdxs = lists:map(fun mango_idx:to_json/1, Idxs),
-	chttpd:send_json(Req, {[{indexes, JsonIdxs}]});
+    JsonIdxs0 = lists:map(fun mango_idx:to_json/1, Idxs),
+    TotalRows = length(JsonIdxs0) ,
+    Limit = case couch_util:get_value(limit, Params, 10000000000) of
```

Comment:
Why not to reuse TotalRows instead of 100000000-a-lot-of-zeroes for default value here?